for-demos/serve_hostname/serve_hostname.go

```diff
@@ -88,8 +90,18 @@ func main() {
 			log.Printf("HTTP request from %s", r.RemoteAddr)
 			fmt.Fprintf(w, "%s", hostname)
 		})
-		go log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", *port), nil))
+		go func() {
+			// Run in a closure so http.ListenAndServe doesn't block
```

Comment:
run via go applies to the first function, but not the arguments.  In this case the http listener is run and blocks (not run via go).  log.Fatal is then run in a go routine with the http listener result passed as an argument.

See this [example old](https://play.golang.org/p/DgVLa-PP2w) - deadlocks
See this [example new](https://play.golang.org/p/-3Mgr2po0N) - passes